test/integration_test/extender_test.go

```diff
@@ -240,3 +246,114 @@ func pvcOwnershipTest(t *testing.T) {
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+
+	logrus.Infof("Waiting for all Pods to come online")
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state")
+
+	scheduledNodes, err := schedulerDriver.GetNodesForApp(ctxs[0])
+	require.NoError(t, err, "Error getting node for app")
+	require.Equal(t, 3, len(scheduledNodes), "App should be scheduled on one node")
+
+	volumeNames := getVolumeNames(t, ctxs[0])
+	require.Equal(t, 1, len(volumeNames), "Should have only one volume")
+
+	logrus.Infof("Verifying Pods scheduling favors antihyperconvergence")
+	verifyAntihyperconvergence(t, scheduledNodes, volumeNames)
+
+	destroyAndWait(t, ctxs)
+}
+
+func antihyperconvergenceTestPreferRemoteOnlyTest(t *testing.T) {
+	ctxs, err := schedulerDriver.Schedule("preferremoteonlytest",
+		scheduler.ScheduleOptions{
+			AppKeys: []string{"test-sv4-svc-repl1-prefer-remote-only"},
+		})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+
+	logrus.Infof("Waiting for all Pods to come online")
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state")
+
+	scheduledNodes, err := schedulerDriver.GetNodesForApp(ctxs[0])
+	require.NoError(t, err, "Error getting node for app")
+	require.Equal(t, 3, len(scheduledNodes), "App should be scheduled on one node")
+
+	volumeNames := getVolumeNames(t, ctxs[0])
+	require.Equal(t, 1, len(volumeNames), "Should have only one volume")
+
+	logrus.Infof("Verifying Pods scheduling favors antihyperconvergence")
+	verifyAntihyperconvergence(t, scheduledNodes, volumeNames)
+
+	logrus.Infof("Cordon the nodes where application pods are running")
+	for _, schedNode := range scheduledNodes {
+		if schedNode.IsStorageDriverInstalled {
+			err = core.Instance().CordonNode(schedNode.Name, defaultWaitTimeout, defaultWaitInterval)
+			require.NoError(t, err, "Error cordorning k8s node for stork test pod")
+			defer func() {
+				err = core.Instance().UnCordonNode(schedNode.Name, defaultWaitTimeout, defaultWaitInterval)
+				require.NoError(t, err, "Error uncordorning k8s node for stork test pod")
+			}()
+		}
+	}
+
+	logrus.Infof("Delete application pods")
+	for _, spec := range ctxs[0].App.SpecList {
+		if obj, ok := spec.(*apps_api.Deployment); ok {
+			depPods, err := apps.Instance().GetDeploymentPods(obj)
+			require.NoError(t, err, "Error getting pods for deployment.")
+
+			err = core.Instance().DeletePods(depPods, true)
+			require.NoError(t, err, "Error delete deployment pods for deploymet.")
+		}
+	}
+
+	logrus.Infof("Waiting for all Pods to come online")
+	err = schedulerDriver.WaitForRunning(ctxs[0], maxPodBringupTime, defaultWaitInterval)
+	require.Error(t, err, "Error waiting for pod to get to running state")
```

Comment:
nit: change the message to "Expected an error while scheduling the pod"

Reply:
ack